### PR TITLE
#12b

### LINKED ISSUE #12: Multipe lines can be wrapped (adjust configuration)
```javascript 
WrapSettings: {
  Delimiters: ' /\_-',
  Labels: [
    {
       Type: 'S1',
       Fields: [
         {
           "Id": 1,
           "Name": "190,69",
           "NrOfLines": 3,
           "MaxChars": 10,
         }
       ]
    }
  ]
}
```

### LINKED ISSUE #17: Test with paths/ Folder

### PR DESCRIPTION
**Pull request by HavermansStef**
- Add CSV export
- Wrap Text logic working as should #3
- Configuration adjusted and working Closes #12
- Added test to check configuration w folder Closes #17

### CODE REVIEW

File: MTX.Conversion.ZebraWrap.App/appsettings.json [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {
+  "Delimiters": "",
+  "Labels": [
+  ],
   "SourcePath": "",
   "TargetPath": "",
-  "WrapTextValues": []
-}
+  "1 = CSV, 2 = Zebra": "",
```
Comment (BartVanHoutte-ADING) [inline MTX.Conversion.ZebraWrap.App/appsettings.json:7]: `"Comments" : "Possible values for OutputMode are (CSV, Zebra)"`

File: MTX.Conversion.ZebraWrap.Contracts/Constants/Configuration/LabelConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+using System.Collections.Generic;
+using MTX.Conversion.ZebraWrap.Contracts.Enums;
+
+namespace MTX.Conversion.ZebraWrap.Contracts.Constants.Configuration
+{
+    public class LabelConfiguration
+    {
+        public LabelTypes Type { get; set; }
+        public List<FieldConfiguration> Fields { get; set; }
```
Comment (BartVanHoutte-ADING) [inline MTX.Conversion.ZebraWrap.Contracts/Constants/Configuration/LabelConfiguration.cs:9]: Add empty line

File: MTX.Conversion.ZebraWrap.Contracts/Constants/WorkListFile.cs [hunk 3]
```diff
@@ -1,21 +1,20 @@
-using System;
 using System.Collections.Generic;
 using MTX.Conversion.ZebraWrap.Contracts.Enums;
 
 namespace MTX.Conversion.ZebraWrap.Contracts.Constants
 {
-    [Serializable]
-    public class WorkListDto
+    public class WorkListFile
```
Comment (BartVanHoutte-ADING) [inline MTX.Conversion.ZebraWrap.Contracts/Constants/WorkListFile.cs:6]: Not a Constants -> Entiteit or Dto

File: MTX.Conversion.ZebraWrap.Contracts/Enums/OutputModes.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+namespace MTX.Conversion.ZebraWrap.Contracts.Enums
+{
+    public enum OutputModes
+    {
+        Invalid = -1,
```
Comment (BartVanHoutte-ADING) [inline MTX.Conversion.ZebraWrap.Contracts/Enums/OutputModes.cs:5]: Invalid and Unknown, both are not processable ... keep one

File: MTX.Conversion.ZebraWrap.Data/Interfaces/Services/IWrapScheduler.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+using System.Collections.Generic;
```
Comment (BartVanHoutte-ADING) [inline MTX.Conversion.ZebraWrap.Data/Interfaces/Services/IWrapScheduler.cs:1]: to remove

File: MTX.Conversion.ZebraWrap.Data/Services/ConversionService.cs [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        #region Private Members
+        
+        private static string ConstructLabelString(LabelToPrint labelToPrint)
+        {
+            var stringBuilder = new StringBuilder();
+            foreach (var field in labelToPrint.Fields)
+            {
+                stringBuilder.AppendLine(field.Line);
+            }
+
```
Comment (BartVanHoutte-ADING) [inline MTX.Conversion.ZebraWrap.Data/Services/ConversionService.cs:76]: `labelToPrint.Fields.ForEach(x => stringBuilder.AppendLine(x.line));`

File: MTX.Conversion.ZebraWrap.Data/Services/WrapScheduler.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
+using System.Collections.Generic;
```
Comment (BartVanHoutte-ADING) [inline MTX.Conversion.ZebraWrap.Data/Services/WrapScheduler.cs:1]: Can be removed

File: MTX.Conversion.ZebraWrap.Tests/BaseTest.cs [hunk 8]
```diff
@@ -25,52 +29,47 @@ public class BaseTest
...
-                        },
-                        new WrapTextValue
-                        {
-                            Label = LabelType.U2,
-                            Lines = 10,
-                            LineId = "145,23",
-                            MaxChars = 10
+                            new FieldConfiguration
+                            {
+                                Id = 0,
```
Comment (BartVanHoutte-ADING) [inline MTX.Conversion.ZebraWrap.Tests/BaseTest.cs:46]: YAGNI

File: MTX.Conversion.ZebraWrap.Contracts/Constants/Configuration/LabelConfiguration.cs (hunk 2 above)
Comment (HavermansStef) [reply MTX.Conversion.ZebraWrap.Contracts/Constants/Configuration/LabelConfiguration.cs:9]: Fixed

File: MTX.Conversion.ZebraWrap.App/appsettings.json (hunk 1 above)
Comment (HavermansStef) [reply MTX.Conversion.ZebraWrap.App/appsettings.json:7]: Fixed

File: MTX.Conversion.ZebraWrap.Contracts/Constants/WorkListFile.cs (hunk 3 above)
Comment (HavermansStef) [reply MTX.Conversion.ZebraWrap.Contracts/Constants/WorkListFile.cs:6]: Fixed

File: MTX.Conversion.ZebraWrap.Contracts/Enums/OutputModes.cs (hunk 4 above)
Comment (HavermansStef) [reply MTX.Conversion.ZebraWrap.Contracts/Enums/OutputModes.cs:5]: Fixed, kept invalid.

File: MTX.Conversion.ZebraWrap.Data/Interfaces/Services/IWrapScheduler.cs (hunk 5 above)
Comment (HavermansStef) [reply MTX.Conversion.ZebraWrap.Data/Interfaces/Services/IWrapScheduler.cs:1]: Fied.

File: MTX.Conversion.ZebraWrap.Data/Services/ConversionService.cs (hunk 6 above)
Comment (HavermansStef) [reply MTX.Conversion.ZebraWrap.Data/Services/ConversionService.cs:76]: Fixed.

**Pull request merged**